phac-nml/irida-next
# Workflow destroy service

## What does this PR do and why?
A user can delete a WorkflowExecution after it has been completed, errored, or canceled.
Fixes #249.

## Screenshots or screen recordings
![Screenshot from 2024-03-05 08-55-47](https://github.com/phac-nml/irida-next/assets/325703/7f723ae7-5257-4e2b-85d6-618484e176a9)
![Screenshot from 2024-03-05 08-55-56](https://github.com/phac-nml/irida-next/assets/325703/252d9af5-b827-4e87-ba76-ae19d3a26e96)
![image](https://github.com/phac-nml/irida-next/assets/325703/307d4a53-fe88-4916-9d19-e47d87560c83)

## How to set up and validate locally
1. Login to irida-next.
2. Create a new workflow execution.
3. Open a rails console and run the following:
```
w = WorkflowExecution.last
w.state = "completed"
w.save
```
5. Refresh the workflow execution page.
6. Click the `Delete` link.
7. Confirm the deletion.
8. Verify the workflow execution has been deleted.

Repeat with `w.state = "error"` and `w.state = "canceled"`.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
